src/Utility/Xml.php

```diff
@@ -203,8 +203,8 @@ protected static function _loadXml($input, $options)
      */
     public static function fromArray($input, $options = [])
     {
-        if (method_exists($input, 'toArray')) {
-            $input = $input->toArray();
+        if (is_object($input) && method_exists($input, 'toArray') && is_callable([$input, 'toArray'])) {
```

Comment:
Is the `is_object()` check necessary? `method_exists()` behaves correctly for scalar values.

```php
php > var_dump(method_exists('nop', 'toArray'));
php shell code:1:
bool(false)
php > var_dump(method_exists(1.1, 'toArray'));
php shell code:1:
bool(false)
```